### PR TITLE
Development

### PR DESCRIPTION
**Pull request by gcammarota**

### CODE REVIEW

**Review by malmans2 — APPROVED** [review]
LGTM!

Unless typer or python console scripts somehow take care of it, the only thing I would change is to make sure we always return to the original working directory, no matter what happens. Something like:
```python
original_cwd = os.getcwd()
os.chdir(run_sub)
try:
    ....
finally:
    os.chdir(original_cwd)
```

**Comment by gcammarota** [comment]
> LGTM!
> 
> Unless typer or python console scripts somehow take care of it, the only thing I would change is to make sure we always return to the original working directory, no matter what happens. Something like:
> 
> ```python
> original_cwd = os.getcwd()
> os.chdir(run_sub)
> try:
>     ....
> finally:
>     os.chdir(original_cwd)
> ```

you're right! and now that you pointed it out, I think I'm gonna split the actual processing from the folders part into a different function

File: c3s_eqc_automatic_quality_control/api.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    # Move into qar subfolder
+    run_sub, qar_id, run_n = _prepare_run_workdir(request, target_dir)
+    os.chdir(run_sub)
+
+    try:
+        run_aqc(request)
+    except Exception:
+        logging.error(traceback.format_exc())
+        logging.error(f"QAR ID: {qar_id} - RUN n.: {run_n} failed ")
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/api.py:163]:
```suggestion
    except Exception:
        logging.exception(f"QAR ID: {qar_id} - RUN n.: {run_n} failed ")
```

Comment (gcammarota) [reply c3s_eqc_automatic_quality_control/api.py]: I think traceback is useful to solve errors but I removed it at this stage of development. I'm gonna add it when implementing the use of logfiles for the dashboard

Comment (gcammarota) [reply c3s_eqc_automatic_quality_control/api.py]: accepted

**Pull request merged**